src/equations/2d/compressible_euler_moist.jl

```diff
@@ -0,0 +1,664 @@
+@doc raw"""
+    CompressibleMoistEnergyEulerEquations2D
```

Comment:
Same argument as below: I think this should be named `CompressibleEulerMoistEnergyEquations2D` or something similar.